src/Elliptic/Executables/Xcts/SolveXcts.hpp

```diff
@@ -0,0 +1,272 @@
+#include "PointwiseFunctions/AnalyticSolutions/Xcts/Flatness.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Xcts/Kerr.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Xcts/Schwarzschild.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/ErrorHandling/FloatingPointExceptions.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace SolveXcts::OptionTags {
+struct NonlinearSolverGroup {
+  static std::string name() noexcept { return "NonlinearSolver"; }
+  static constexpr Options::String help = "The iterative nonlinear solver";
+};
+struct NewtonRaphsonGroup {
+  static std::string name() noexcept { return "NewtonRaphson"; }
+  static constexpr Options::String help =
+      "Options for the Newton-Raphson nonlinear solver";
+  using group = NonlinearSolverGroup;
+};
+struct LinearSolverGroup {
+  static std::string name() noexcept { return "LinearSolver"; }
+  static constexpr Options::String help =
+      "The iterative Krylov-subspace linear solver";
+};
+struct GmresGroup {
+  static std::string name() noexcept { return "Gmres"; }
+  static constexpr Options::String help = "Options for the GMRES linear solver";
+  using group = LinearSolverGroup;
+};
+}  // namespace SolveXcts::OptionTags
+
+/// \cond
+struct Metavariables {
+  static constexpr size_t volume_dim = 3;
+  static constexpr int conformal_matter_scale = 0;
+  using system =
+      Xcts::FirstOrderSystem<Xcts::Equations::HamiltonianLapseAndShift,
+                             Xcts::Geometry::Curved, conformal_matter_scale>;
+
+  // List the possible backgrounds, i.e. the variable-independent part of the
+  // equations that define the problem to solve (along with the boundary
+  // conditions)
+  using analytic_solution_registrars =
+      tmpl::list<Xcts::Solutions::Registrars::Schwarzschild,
+                 Xcts::Solutions::Registrars::Kerr>;
+  using analytic_data_registrars = tmpl::list<
+      Xcts::AnalyticData::Registrars::Binary<analytic_solution_registrars>>;
+  using background_tag = elliptic::Tags::Background<
+      ::AnalyticData<volume_dim, tmpl::append<analytic_solution_registrars,
+                                              analytic_data_registrars>>>;
+
+  // List the possible initial guesses
+  using initial_guess_registrars =
+      tmpl::append<tmpl::list<Xcts::Solutions::Registrars::Flatness>,
+                   analytic_solution_registrars, analytic_data_registrars>;
+  using initial_guess_tag = elliptic::Tags::InitialGuess<
+      ::AnalyticData<volume_dim, initial_guess_registrars>>;
+
+  static constexpr Options::String help{
+      "Find the solution to an XCTS problem."};
+
+  // These are the fields we solve for
+  using fields_tag = ::Tags::Variables<typename system::primal_fields>;
+  // These are the fluxes corresponding to the fields, i.e. essentially their
+  // first derivatives. These are background fields for the linearized sources.
```

Comment:
just checking: are these both the fluxes and the background fields for the linearized sources?

Reply:
yes. These are the fluxes for the nonlinear DG operator. They act as background fields for the linearized DG operator.